packages/webapp/src/containers/bulkSensorReadingsSlice.js

```diff
@@ -13,64 +12,74 @@ const initialState = {
   activeReadingTypes: [],
 };
 
+const sensorReadingAdapter = createEntityAdapter({
+  selectId: (sensorReading) => sensorReading.reading_id,
+});
+
 const bulkSensorsReadingsSlice = createSlice({
   name: 'bulkSensorsReadingsReducer',
-  initialState,
+  initialState: sensorReadingAdapter.getInitialState({
+    loading: false,
+    error: undefined,
+    loaded: false,
```

Comment:
Yes. I took a look at other slices and this seemed to be the pattern used. These should help the slice recognize state changes and re-render the frontend properly.

Reply:
Sorry, what I meant was can you add loading, error, and loaded to `initialState` on line 3? Or did you mean they were not recognized if you did so?

Reply:
I added those patterns but never used them :sweat_smile: (copied from apollo)
I never figured out a way to solve the store hydration order problem such as the blank task page bug.
If task/management plan/crop slices are hydrated in random order, task selectors will break since associated management plans might not be in store when tasks are fetched.

To patch the issue, redux store is hydrated from leaf to root in a waterfall style, which is terrible.
crop -> crop_variaty->management_plan->tasks

My initial thought was to use loading in selectors. In taskSelector can return an empty array if dependencies are in a loading state. Manually managing the loading state is a pain. The approach can't really eliminate the broken selector problem.

Another solution to broken selector problem is for endpoints to return the entire object.
For example, task endpoints will return task -> management plans -> crop_variety -> crop
->locations. But the request will be very big, and the response needs to be gzipped. 

It would be great if you guys can figure out the hydration problem.

Reply:
So what I did here was revert all changes to existing. I think I tried to do more cleanup than necessary. Need to learn more about these.